docs/source/model-registry.rst

```diff
@@ -78,47 +79,169 @@ version to one of the other valid stages.
+
+   client = MlflowClient()
+   client.create_registered_model("sk-learn-random-forest-reg-model")
+
+While the method above creates an empty registered model with no version associated, the method below creates a new version of the model.
 
 .. code-block:: py
 
-   mlflow.tensorflow.log_model(tf_saved_model_dir=saved_estimator_path,
-                               tf_meta_graph_tags=[tag_constants.SERVING],
-                               tf_signature_def_key="predict",
-                               artifact_path="model",
-                               registered_model_name="EmailMarketingCampaign")
+    client = MlflowClient()
+    result = client.create_model_version(name="sk-learn-random-forest-reg-model",
+                source="mlruns/0/d16076a3ec534311817565e6527539c0/artifacts/sk-model",
+                run_id="d16076a3ec534311817565e6527539c0")
+
+In contrast, ``mlflow.register_model()`` and ``mlflow.<model_flavor>.log_model()`` will create a new version in the registry, if it does not already exist.
 
-.. note:: If the registered model does not already exist, the registry will create one before creating a new model version.
+Adding or Updating Model Descriptions
+-------------------------------------
 
-To confirm that the model has successfully been registered, call the ``get_model_version_details`` API:
+At any point in a model’s lifecycle development, you can update a model version's description using the MLflow Tracking API.
 
 .. code-block:: py
 
-   client = mlflow.tracking.MlflowClient()
-   client.get_model_version_details(name = "EmailMarketingCampaign",
-                                    version = 12)
+    client = MlflowClient()
+    client.update_model_version(name="sk-learn-random-forest-reg-model",
+            version=1,
+            description="This model version is a scikit-learn random forest containing 100 decision trees")
 
-You can query for the latest version of each registered model by stage using the ``get_latest_versions`` API:
+As well as adding or updating a description of a specific version of the model, you can rename an existing registered model.
 
 .. code-block:: py
 
-   client.get_latest_versions(name = "EmailMarketingCampaign",
-                              stages = ["Staging"])
+    client=MlflowClient()
+    client.rename_registered_model(name="sk-learn-random-forest-reg-model",
+            new_name="sk-learn-random-forest-reg-model-100")
+
+Transitioning an MLflow Model’s Stage
+-------------------------------------
+Over the course of the model’s lifecycle, a model evolves—from development to staging to production.
+You can transition a registered model in the registry to one of the stages: **None**, **Staging**, **Production** or **Archived**.
+
+.. code-block:: py
+
+    client = MlflowClient()
+    client.transition_model_version_stage(name="sk-learn-random-forest-reg-model",
+  	        version=3,
+	        stage="production")
+
+Listing and Searching MLflow Registered Models
```

Comment:
=> Listing and Searching Models

(because `list` an API on registered models and `search` on model versions)

Reply:
I'm on it. Will commit shortly.

Reply:
done and addressed it with a new set of commits